app/lib/search/updater.dart

```diff
@@ -37,6 +37,26 @@ class BatchIndexUpdater implements TaskRunner {
   // start do the index merges, making sure that the index is marked as ready.
   int _firstScanCount;
 
+  /// Loads the package index snapshot, or if it fails, creates a minimal
+  /// package index with only package names and minimal information.
+  Future init() async {
+    await _initSnapshot();
+
+    if (!packageIndex.isReady) {
+      _logger.info('Loading minimum package index...');
+      int cnt = 0;
+      await for (final pd in searchBackend.loadMinimumPackageIndex()) {
+        await packageIndex.addPackage(pd);
+        cnt++;
+        if (cnt % 500 == 0) {
+          _logger.info('Loaded $cnt minimum package data (${pd.package})');
+        }
+      }
+      await packageIndex.merge();
```

Comment:
As dig into what `merge()` and `isReady` means... Maybe these are things we should document?

I know that sort of outside the scope of this PR..

Reply:
These should be refactored (on my list, but very low priority), as the merge is not really a merge with the current library...